### ISSUE TITLE
Possible Endless Loop

### ISSUE THREAD

**Issue by megel** [HodorNV/ALOps#24]
If not enough memory is available, this task loop endless:
![image](https://user-images.githubusercontent.com/15800069/64014311-1a2d1700-cb22-11e9-9fc6-8610d878f00e.png)

**Comment by AdminHodor** [comment]
Hello,

We checked  this out  and where able to replicate this, the loop is maintained in the powershell running inside the container. We cannot modify this from ALOps. 

We will however make a change to ALOpsDockerWait method so it can recognize this scenario and force-quite the container. This should mitigate the endless looping and eating up build-time.

**Comment by AdminHodor** [comment]
Hello @megel ,

We did manage to setup a test-scenario for this error just by running a build on a swamped agent. 
Modifications are made to detect this specific error and cancel the build if it occurs, not using up any additional build-time. 
Furthermore we added global-variables which can be used to monitor available resources of the agent while running the pipeline. By setting "Alops_PrintMemoryStats" = 'true' in the pipeline variables you will (with the current release) get a printout of resource usage in every step. Improvement for this is also in the pipeline, where we will provide extra variables to setup limits which will trigger a warning in the pipeline if met.

Kind regards,

Bart Bourgeois
